node/node_cross_shard.go

```diff
@@ -64,6 +64,33 @@ func (node *Node) BroadcastCXReceipts(newBlock *types.Block, lastCommits []byte)
 	}
 }
 
+// BroadcastMissingCXReceipts broadcasts missing cross shard receipts per request
+func (node *Node) BroadcastMissingCXReceipts() {
+	sendNextTime := []common.Hash{}
+	it := node.CxPool.Pool().Iterator()
+	for blockHash := range it.C {
+		blk := node.Blockchain().GetBlockByHash(blockHash.(common.Hash))
```

Comment:
Please do not use `GetBlockByHash` and later `GetBlockByNumber` just to use its header.  Use `GetHeaderByHash` and `GetHeaderByNumber` instead.

Reply:
good point. changed